File: JAVA/src/main/java/models/Partner.java
package models;

import java.util.Date;

public class Partner {
    private int idPartner;
    private String namePartner;
    private String typePartner;
    private String description;
    private String email;
    private int tel;
    private String image;

    public Partner() {
    }

    public Partner(int idPartner, String namePartner, String typePartner, String description, String email, int tel,String image) {
        this.idPartner = idPartner;
        this.namePartner = namePartner;
        this.typePartner = typePartner;
        this.description = description;
        this.email = email;
        this.tel = tel;
        this.image = image;
    }

    public Partner( String namePartner, String typePartner, String description, String email, int tel,String image) {
        this.namePartner = namePartner;
        this.typePartner = typePartner;
        this.description = description;
        this.email = email;
        this.tel = tel;
        this.image = image;
    }
    public Partner(int idPartner, String namePartner) {
        this.idPartner = idPartner;
        this.namePartner = namePartner;
    }

    public int getIdPartner() {
        return idPartner;
    }


    public String getNamePartner() {
        return namePartner;
    }

    public String getTypePartner() {
        return typePartner;
    }

    public String getDescription() {
        return description;
    }

    public String getEmail() {
        return email;
    }

    public int getTel() {
        return tel;
    }

    public void setIdPartner(int idPartner) {
        this.idPartner = idPartner;
    }


    public void setNamePartner(String namePartner) {
        this.namePartner = namePartner;
    }

    public void setTypePartner(String typePartner) {
        this.typePartner = typePartner;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public void setTel(int tel) {
        this.tel = tel;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    @Override
    public String toString() {
        return "Partner{" +
                "idPartner=" + idPartner +
                ", namePartner='" + namePartner + '\'' +
                ", typePartner='" + typePartner + '\'' +
                ", description='" + description + '\'' +
                ", email='" + email + '\'' +
                ", tel=" + tel +
                ", image='" + image + '\'' +
                '}';
    }
}

File: Symfony/src/Entity/Partner.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use App\Repository\PartnerRepository;
use Symfony\Component\Serializer\Annotation\Groups;


#[ORM\Entity(repositoryClass: PartnerRepository::class)]
class Partner
{
    
    #[ORM\Id]
    #[ORM\GeneratedValue(strategy: "AUTO")]
    #[ORM\Column(name: "idPartner", type: "integer")]
    #[Groups(['event:read'])]
    private ?int $idpartner;

    #[ORM\Column(length :255)]
    #[Groups(['event:read'])]
    private ?string $namepartner;

    #[ORM\Column(length :255)]
    #[Groups(['event:read'])]
    private ?string $typepartner;

    #[ORM\Column(length :255)]
    #[Groups(['event:read'])]
    private ?string $email;

    #[ORM\Column]
    #[Groups(['event:read'])]
    private ?int $tel;

    #[ORM\Column(length :255)]
    #[Groups(['event:read'])]
    private ?string $description;

    #[ORM\Column(length :255)]
    #[Groups(['event:read'])]
    private ?string $image;

    public function getIdpartner(): ?int
    {
        return $this->idpartner;
    }

    public function getNamepartner(): ?string
    {
        return $this->namepartner;
    }

    public function setNamepartner(string $namepartner): static
    {
        $this->namepartner = $namepartner;

        return $this;
    }

    public function getTypepartner(): ?string
    {
        return $this->typepartner;
    }

    public function setTypepartner(string $typepartner): static
    {
        $this->typepartner = $typepartner;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): static
    {
        $this->email = $email;

        return $this;
    }

    public function getTel(): ?int
    {
        return $this->tel;
    }

    public function setTel(int $tel): static
    {
        $this->tel = $tel;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(string $image): static
    {
        $this->image = $image;

        return $this;
    }


}
